core/database_api.php

```diff
@@ -59,6 +59,9 @@
 $ADODB_FETCH_MODE = ADODB_FETCH_ASSOC;
 define( 'ADODB_ASSOC_CASE', ADODB_ASSOC_CASE_LOWER );
 
+# Stores the functional database type based on db driver
+$g_db_functional_type;
```

Comment:
Undefined at that point. But present for the sake of documenting it.
This could be initialized there, or as it is now, from db_connect()